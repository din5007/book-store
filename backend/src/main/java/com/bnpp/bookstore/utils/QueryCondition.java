package com.bnpp.bookstore.utils;

import io.micrometer.common.util.StringUtils;
import org.springframework.stereotype.Component;

@Component("queryUtil")
public class QueryCondition {

  public boolean checkNull(String field) {
    return StringUtils.isBlank(field);
  }
}
