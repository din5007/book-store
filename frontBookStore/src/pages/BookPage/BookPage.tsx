import BookCard from "../../components/Card/Card";
import { FC } from "react";
import SearchBook from "../../components/SearchBook/Search";
import CustomPagination from "../../components/Pagination/Pagination";

const BookPage: FC<{}> = () => {
  return (
    <div
      className="d-flex align-items-center flex-column"
      style={{ margin: "5%" }}
    >
      <SearchBook />
      <BookCard />
      <CustomPagination />
    </div>
  );
};
export default BookPage;
