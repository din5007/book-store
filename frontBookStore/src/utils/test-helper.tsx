import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";


function customRender(component : any , store?: any) {

    return render(<StoreProvider store={store}>
        <BrowserRouter>
            {component}
        </BrowserRouter>
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;