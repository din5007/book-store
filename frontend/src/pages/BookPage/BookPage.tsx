import { Card } from "react-bootstrap";
import BookCard from "../../components/Card/Card";
import { FC, useState } from "react";
import SearchBook from "../../components/SearchBook/Search";

const BookPage : FC<{}> = ()=> {
    const [ searchText , setSearchText ] = useState('');

    return (
        <div className="d-flex align-items-center flex-column" style={{ margin : "5%"}}>
            <SearchBook onChange={(e : any) => setSearchText(e?.target?.value)}/>
            <BookCard />
        </div>
    )
}
export default BookPage;