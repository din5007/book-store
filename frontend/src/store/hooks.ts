import { createTypedHooks } from "easy-peasy";

import { AppStoreModel } from ".";

export const { useStoreActions, useStoreState } = createTypedHooks<AppStoreModel>();