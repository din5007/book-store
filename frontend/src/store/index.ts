import { HttpService } from "../service/http/http.service";
import { createStore } from 'easy-peasy';
import { BookModel, booksModel } from "./books/books.model";
import { cartModel, CartModel } from "./cart/cart.model";
import { userModel, UserModel } from "./users/user.model";
import { OrderModel, orderModel } from "./order/order.model";

export interface AppStoreModel {
    books: BookModel;
    cart: CartModel;
    user: UserModel;
    order: OrderModel;
}

export const appStoreModel : AppStoreModel = {
    books: booksModel,
    cart: cartModel,
    user: userModel,
    order: orderModel,
}

export function storeModel() {
    return createStore(appStoreModel, {
        injections: {
            httpService: new HttpService()
        },
        initialState: {
            
        },
    })
}
export const store =  storeModel();