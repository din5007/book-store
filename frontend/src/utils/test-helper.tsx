import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { AppStoreModel, store } from "../store";
import { Router } from 'react-router-dom';
import { ReactNode } from "react";



// const provider : React.FC<{children : ReactNode, store: any}> = ({ children, store }) => {
//     return (
        
//     )
// }

function customRender(component : any , store: any) {

    return render(<StoreProvider store={store}>
        {component}
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;