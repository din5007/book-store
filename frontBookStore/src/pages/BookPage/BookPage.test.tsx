import { createStore } from 'easy-peasy';
import { customRender as render, waitFor } from '../../utils/test-helper';
import BookPage from "./BookPage"
import { appStoreModel } from '../../store';
import { describe, expect, it, vi } from 'vitest';

const mock = {
    get: vi.fn()
};

const store = createStore(appStoreModel, {
    injections: {
        httpService : mock
    },
    initialState: {
        books : {
            books: [{ id : 1, title : 'Harry Potter', author: 'JK rowling', price: 400, quantity: 400}]
        }
    }
})

describe('Book Page', () => {
    it('Book Page should load properly - snapshot', async () => {
        const {container}  = render(<BookPage/>, store);
        waitFor(() => {
            expect(container).toMatchSnapshot();
        });
    });

    it('Book Page should contain Search component', async () => {
        const {getByPlaceholderText}  = render(<BookPage/>, store);
        waitFor(() => {
            expect(getByPlaceholderText('Search by Book Name')).toBeDefined();
        })
    });
})