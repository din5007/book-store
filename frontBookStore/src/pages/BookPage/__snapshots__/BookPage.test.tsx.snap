// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Book Page > Book Page should load properly - snapshot 1`] = `
<div>
  <div
    class="d-flex align-items-center flex-column"
    style="margin: 5%;"
  >
    <div
      class="mb-3"
    >
      <input
        class="form-control"
        id="exampleForm.ControlInput1"
        placeholder="Search by Book Name"
        type="search"
      />
    </div>
    <div
      class="g-4 m-auto row row-cols-lg-4 row-cols-md-3 row-cols-1"
    >
      <div
        class="col"
      >
        <div
          class="card"
          style="width: 18rem;"
        >
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              HARRY POTTER
            </div>
            <p
              class="card-text"
            >
              JK ROWLING
            </p>
            <p
              class="card-text"
            >
              Price : 
              400
               INR
            </p>
            <button
              class="btn btn-primary"
              type="button"
            >
              Add to Cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
